src/imgui/imgui_raii_wrapper.h

```diff
@@ -0,0 +1,162 @@
+#ifndef MYPROJECT_IMGUI_RAII_WRAPPER_H
+#define MYPROJECT_IMGUI_RAII_WRAPPER_H
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <imgui.h>
+
+namespace irw {
+
+class Windowzita {
+public:
+  [[nodiscard]] explicit Windowzita(std::string const& id) : selected{ImGui::Begin(id.c_str())} {}
+  Windowzita(Windowzita const&) = delete;
+  Windowzita& operator=(Windowzita const&) = delete;
+  Windowzita(Windowzita&&) = delete;
+  Windowzita& operator=(Windowzita&&) = delete;
+  ~Windowzita() {
+    ImGui::End();
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabBar {
+public:
+  [[nodiscard]] explicit TabBar(std::string const& id) : selected{ImGui::BeginTabBar(id.c_str())} {}
+  TabBar(TabBar const&) = delete;
+  TabBar& operator=(TabBar const&) = delete;
+  TabBar(TabBar&&) = delete;
+  TabBar& operator=(TabBar&&) = delete;
+  ~TabBar() {
+    if (selected) {
+      ImGui::EndTabBar();
+    }
+  }
```

Comment:
This comment is valid for all of the destructors.
I really think there's no strong reasoning to continue to build something when `ImGui::BeginWtvThing` returns `false`. 
I guess that means that `ImGui` wasn't able to construct the GUI object properly, so imho we should just stop the object construction there.